agent/command/papertrail_trace.go

```diff
@@ -73,12 +68,61 @@ func (t *papertrailTrace) Execute(ctx context.Context,
 			return errors.Wrap(err, "running trace")
 		}
 
-		logger.Task().Infof("Successfully traced '%s' (sha256://%s)", fullname, sha256sum)
+		logger.Task().Infof("Successfully traced '%s' (sha256://%s)", f.filename, f.sha256)
 	}
 
 	return nil
 }
 
+type papertrailTraceFile struct {
+	filename string
+	sha256   string
+}
+
+func getTraceFiles(workdir string, patterns []string) ([]papertrailTraceFile, error) {
+	files := make([]papertrailTraceFile, 0, len(patterns))
+
+	seen := make(map[string]string)
+
+	for _, path := range patterns {
+		patternpath := filepath.Join(workdir, path)
+
+		matches, err := filepath.Glob(patternpath)
+		if err != nil {
+			return nil, errors.Wrap(err, "globbing filepath")
+		}
+
+		for _, matchpath := range matches {
+			sha256sum, err := getSHA256(matchpath)
+			if err != nil {
+				return nil, errors.Wrap(err, "getting sha256")
+			}
+
+			basename := filepath.Base(matchpath)
+
+			// santity check that the filename we're tracing is legitimate
```

Comment:
By "legitimate" here, do we mean it's a file and not a directory? If so, can we clarify that only files are legitimate, and not directories?